luaui/Widgets/unit_smart_area_reclaim.lua

```diff
@@ -169,147 +171,184 @@ local function issue(rList, shift)
 	end
 end
 
+local reclaimOrders = {}
+
+-- we use the previous unit loop iterating on cmds to store reclaim orders
+-- and return cmds for its original usage
+local function storeReclaimOrders(uid)
+		local cmds = GetUnitCommands(uid, maxOrdersCheck)
+
+		reclaimOrders[uid] = {}
+		local reclaimOrdersCount = 0
+
+		for _, order in pairs(cmds) do
+			if order["id"] == RECLAIM then
+				reclaimOrdersCount = reclaimOrdersCount + 1
+				reclaimOrders[uid][reclaimOrdersCount] = order["params"][1]
+			end
+		end
+
+		return cmds
+end
+
+local function checkNoDuplicateOrder(uid, fid)
+		local orderParam = fid+maxUnits
+
+		for _, reclaimParam in pairs(reclaimOrders[uid] or {}) do
+			if reclaimParam == orderParam then
+				return false
+			end
+		end
+
+		return true
+end
+
 function widget:CommandNotify(id, params, options)
-	if id == RECLAIM then
-		local mobiles, stationaries = {}, {}
-		local mobileb, stationaryb = false, false
-
-		local rUnits = {}
-		local rUnitsCount = 0
-		local sUnits = GetSelectedUnits()
-		for i=1, #sUnits do
-			local uid = sUnits[i]
-			local udid = GetUnitDefID(uid)
-			if unitCanReclaim[udid] then
-				if not unitCanMove[udid] then
-					stationaries[uid] = unitBuildDistance[udid]
-					stationaryb = true
-				else
-					mobiles[uid] = unitBuildDistance[udid]
-					mobileb = true
-				end
+	if id ~= RECLAIM then return false end
```

Comment:
Unnesting, it becomes a mess to review after this but I'll highlight the most relevant sections below.